flytepropeller/pkg/controller/nodes/task/handler.go

```diff
@@ -435,15 +437,38 @@ func (t Handler) fetchPluginTaskMetrics(pluginID, taskType string) (*taskMetrics
 	if err != nil {
 		return nil, err
 	}
-	if _, ok := t.taskMetricsMap[metricNameKey]; !ok {
-		t.taskMetricsMap[metricNameKey] = &taskMetrics{
-			taskSucceeded: labeled.NewCounter(metricNameKey+"_success",
-				"Task "+metricNameKey+" finished successfully", t.pluginScope, labeled.EmitUnlabeledMetric),
-			taskFailed: labeled.NewCounter(metricNameKey+"_failure",
-				"Task "+metricNameKey+" failed", t.pluginScope, labeled.EmitUnlabeledMetric),
-		}
+
+	// Acquire read lock for fast read, this is the happy case
+	t.taskMetricsMapMutex.RLock()
```

Comment:
After startup there will likely be no writes to this map so using a shared read lock is a minor optimization so there is no lock contention.